### PR TITLE
Inline server side factory objects

### PR DESCRIPTION
**Pull request by DanVanAtta**
Refactor of http-server objects and how they are built.

To simplify and hide dependencies, delete single method factory classes
and move factory methods to the class that is being built. Second, avoid
building nested dependencies and add a factory build method to dependencies.

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[x] Code Cleanup or refactor
[] Configuration Change
[] Problem fix:  
[] Other:

### CODE REVIEW

**Comment by ron-murhammer** [comment]
@DanVanAtta Does this potentially have conflicts with the websocket PR? I see its touching some of the web socket classes.

File: http-server/src/main/java/org/triplea/http/ServerApplication.java [hunk 1]
```diff
@@ -165,7 +160,7 @@ public void run(final AppConfig configuration, final Environment environment) {
         .getUserProperties()
         .put(PlayerConnectionWebSocket.GAME_LISTING_QUEUE_KEY, gameListingEventQueue);
 
-    final var messagingService = MessagingServiceFactory.build(jdbi, chatSessions, chatters);
+    final var messagingService = ChatMessagingService.build(jdbi, chatSessions, chatters);
```
Comment (ron-murhammer) [inline http-server/src/main/java/org/triplea/http/ServerApplication.java]: Its not clear to me if the variable type changed here? Should the var name be updated to `chatMessagingService`?

File: http-server/src/main/java/org/triplea/modules/error/reporting/ErrorReportController.java [hunk 2]
```diff
@@ -11,17 +12,38 @@
...
+  public static ErrorReportController build(final AppConfig configuration, final Jdbi jdbi) {
+    final GithubIssueClient githubIssueClient =
+        GithubIssueClient.builder()
+            .uri(AppConfig.GITHUB_WEB_SERVICE_API_URL)
+            .authToken(configuration.getGithubApiToken())
+            .githubOrg(AppConfig.GITHUB_ORG)
+            .githubRepo(configuration.getGithubRepo())
+            .build();
+
+    if (githubIssueClient.isTest()) {
```
Comment (ron-murhammer) [inline http-server/src/main/java/org/triplea/modules/error/reporting/ErrorReportController.java]: Can you clarify what this block is for?

File: http-server/src/main/java/org/triplea/modules/game/ConnectivityController.java [hunk 3]
```diff
@@ -26,6 +26,12 @@
 public class ConnectivityController extends HttpController {
   @Nonnull private final Predicate<InetSocketAddress> connectivityCheck;
 
+  public static ConnectivityController build() {
+    return ConnectivityController.builder() //
```
Comment (ron-murhammer) [inline http-server/src/main/java/org/triplea/modules/game/ConnectivityController.java]: Is the `//` for formatting reasons?

**Comment by DanVanAtta** [comment]
> @DanVanAtta Does this potentially have conflicts with the websocket PR? I see its touching some of the web socket classes.

No, but it conflicts with https://github.com/triplea-game/triplea/pull/6096, and that does conflict with the websocket technology change. As long as we keep the same abstraction  in GenericWebsocketListener, we'll be able to integrate the two.

**Comment by ron-murhammer** [comment]
@DanVanAtta Ok. Well I think you and @RoiEXLab need to decide how to move forward on those then (I'm fine being the deciding vote if need be). If this doesn't conflict with either of them then feel free to merge it.

File: http-server/src/main/java/org/triplea/http/ServerApplication.java (hunk 1 above)
Comment (DanVanAtta) [reply http-server/src/main/java/org/triplea/http/ServerApplication.java]: It's not a super critical update as `ChatMessagingService` is still a 'messagingService'. `MessagingServiceFactory` was renamed as it was too generic, it really was 'chat' messages, and *not* things like network communication, internal events, queues, etc..

Let's keep an eye on this for the branch in #6096, updating this here would conflict with that, that is my only hesitation to not update it now.

The main thrust of this PR is that we are going from factory classes, to a factory method.

File: http-server/src/main/java/org/triplea/modules/error/reporting/ErrorReportController.java (hunk 2 above)
Comment (DanVanAtta) [reply http-server/src/main/java/org/triplea/modules/error/reporting/ErrorReportController.java]: It is moved code. I can't find where @RoiEXLab had a similar question, that answer would have been useful here.

In short it's a validation check of configuration to ensure we do not have an 'issue client' set to a 'test' mode with a prod configuration set. 

The `isTest` check is defined as:

```
  public boolean isTest() {
    return authToken.equalsIgnoreCase("test");
  }
```

In short we get a stubbing behavior when the authtoken used by the server to create github issues is set to 'test'.

File: http-server/src/main/java/org/triplea/modules/game/ConnectivityController.java (hunk 3 above)
Comment (DanVanAtta) [reply http-server/src/main/java/org/triplea/modules/game/ConnectivityController.java]: Indeed

**Pull request merged**